src/reset.c

```diff
@@ -0,0 +1,90 @@
+#include "reset.h"
+#include "asic_types.h"
+#include "stdint.h"
+#include "uprintf.h"
+
+#include "netx_io_areas.h"
+
+// Configure watchdog register base address for each chip type
+#if   ASIC_TYP==ASIC_TYP_NETX500
+static const uint32_t ulWdgBaseAddr = NX500_NETX_WDG_AREA;
+#elif ASIC_TYP==ASIC_TYP_NETX50
+static const uint32_t ulWdgBaseAddr = NX50_NETX_WDG_AREA;
+#elif ASIC_TYP==ASIC_TYP_NETX10
+static const uint32_t ulWdgBaseAddr = NX10_NETX_WDG_AREA;
+#elif ASIC_TYP==ASIC_TYP_NETX56
+static const uint32_t ulWdgBaseAddr = NX56_NETX_WDG_AREA;
+#elif ASIC_TYP==ASIC_TYP_NETX4000_RELAXED
+static const uint32_t ulWdgBaseAddr = NX4000_NETX_WDG_AREA;
+#elif ASIC_TYP==ASIC_TYP_NETX4000
+static const uint32_t ulWdgBaseAddr = NX4000_NETX_WDG_AREA;
+#elif ASIC_TYP==ASIC_TYP_NETX90_MPW
+static const uint32_t ulWdgBaseAddr = Addr_NX90MPW_wdg_com;
+#elif ASIC_TYP==ASIC_TYP_NETX90
+static const uint32_t ulWdgBaseAddr = Addr_NX90_wdg_com;
+#elif ASIC_TYP==ASIC_TYP_NETIOL
+static const uint32_t ulWdgBaseAddr = Addr_NIOL_wdg_sys;
+#else
+static const uint32_t ulWdgBaseAddr = 0;
+#warning "This netX type does not support resets."
+#endif
+
+NETX_CONSOLEAPP_RESULT_T resetNetX(void){
+    #if (ASIC_TYP==ASIC_TYP_NETX4000 || ASIC_TYP==ASIC_TYP_NETX4000_RELAXED)
+    //These require special treatment and are not implemented yet
+    uprintf("Error: This hardware is not supported");
+    return NETX_CONSOLEAPP_RESULT_ERROR;
+    #elif (ASIC_TYP==ASIC_TYP_NETIOL)
```

Comment:
This procedure (for the watchdog on netIOL) differs quite a bit from the implementation in flasher_helper.lua. It should be tested.

Reply:
Copied your implementation over into C